### ISSUE TITLE
error Google Drive's desktop.ini files

### ISSUE THREAD

**Issue by alemacedo** [mhutchie/vscode-git-graph#611]
**Description**
As I use Google Drive app, it keeps generating desktop.ini files in folders. It seems that it's crashing the plugin.

**Steps to Reproduce**
1. Just use Google Drive syncing in some folder where is a Git project.
2. Try to open the plugin to visualize the branches and it crashes.
3. It shows the message: "Error: Unable to load Commits. fatal: bad object desktop.ini"
4. I've done a test and deleted all desktop.ini files and the plugin got back working.

**Environment**
 - Git Graph Extension Version: v1.30.0
 - Visual Studio Code Version: v1.63.2
 - Operating System: Windows 11
 - Google Drive app: v54.0.2.0

**Screenshots**
![image](https://user-images.githubusercontent.com/26600950/148590844-853a19b1-527a-44b2-939e-32e5946da85b.png)

**Comment by mhutchie** [comment]
Hi @alemacedo,

Git Graph uses the Git executable installed on your computer to run all Git commands against a repository. The error message you reference is a fatal error thrown by Git indicating that the repository (on your file system) is not in a valid state. It's unrelated to the extension - if you run the same Git command in a terminal, you'd get the same error.

When Git is unable to provide the requested commit information to the extension, the extension displays the error message to the user so they can resolve the underlying Git error. The extension can't be expected to do anything else if Git throws a fatal error, as it doesn't have the necessary information to render the graph.

FYI: It's recommended to use the native Git "remote" capabilities for backing up or sharing your repository, as they are specifically designed for this purpose. In the majority of Git repositories, Git will create and manage thousands of files in the `.git` directory of your repository to store the commits, branches, and tags. Unless an exact replica of all these files a transferred in the correct order, it's quite likely for the repository to be put into a temporary of permanently invalid state. Google Drive & other file backup tools are not well suited for this use case. In your situation it seems Google Drive is creating desktop.ini files in internal Git directories where they are definitely not allowed (hence the fatal error thrown by Git).

**Comment by shehzadaliqadri** [comment]
So, what's is the solution of unwanted desktop.ini files that are creating in .git folder by google drive. is there any solution for restricting google drive not to create .desktop.ini files in .git folder or is there any need for github to handle this prescribed issue.

**Comment by alemacedo** [comment]
Hi @mhutchie!

Thank you so much for your attention.
For all the purposes, I've followed your tip, and now I'm not using Drive to backup my git project folders (way more clever, haha), especially because of the "node_modules" folders, as the Drive hasn't a way to prevent the sync of them.

But, before stop using Drive in git project folders, the interesting part is that I've inserted the desktop.ini files in .gitignore, and the git works fine for all other use cases, except for the extension.

Anyway, it's not a problem for me anymore.
Thanks.

**Comment by alemacedo** [comment]
> So, what's is the solution of unwanted desktop.ini files that are creating in .git folder by google drive. is there any solution for restricting google drive not to create .desktop.ini files in .git folder or is there any need for github to handle this prescribed issue.

So far, Drive hasn't a way to prevent the creation of desktop.ini files, plus Drive does sync "node_modules" folders/files, for example, and it's not a good idea.

As @mhutchie said, it's probably the best solution to not use Drive in a git project folder.

I've created a code in python to delete files in batch. You can check it here: https://github.com/alemacedo/python-threading-delete_file.
Proceed with caution, and you can delete all desktop.ini files in a given folder. The extension will work again. But at some point, Drive will create desktop.ini files again...

**Comment by spsquared** [comment]
Slightly unrelated here, but I can't fetch the repository because of the random desktop.ini files. I also can't find them so I can't delete them, even though Windows says they exist. It won't let me create a file called "desktop.ini" because "there is already a file with the same name in this location." I can't even go look at the repository without getting errors.

**Comment by Atanu17** [comment]
> Slightly unrelated here, but I can't fetch the repository because of the random desktop.ini files. I also can't find them so I can't delete them, even though Windows says they exist. It won't let me create a file called "desktop.ini" because "there is already a file with the same name in this location." I can't even go look at the repository without getting errors.

Same problem!

**Comment by sharif-smj** [comment]
First and foremost. dont sync your project files in google drive as it creates these desktop.ini files in every directory to show you that icon that tells you that the folder is synced or shared in drive.

I used the script below to solve this problem. It uses the os module to traverse all subdirectories of the given folder and delete any file named desktop.ini. It will print a message for each file that it successfully deletes or encounters an error while attempting to delete.


import os

def delete_desktop_ini_files(folder_path):
    for root, dirs, files in os.walk(folder_path):
        for filename in files:
            if filename.lower() == "desktop.ini":
                file_path = os.path.join(root, filename)
                try:
                    os.remove(file_path)
                    print(f"Deleted {file_path}")
                except Exception as e:
                    print(f"Error deleting {file_path}: {e}")

if __name__ == "__main__":
    folder_path = input("Enter the folder path: ")
    delete_desktop_ini_files(folder_path)




Open the Command Prompt or PowerShell in Windows.
Navigate to the directory where the script is saved.
Type python script_name.py in the command line, replacing script_name.py with the name of your Python script.
Press Enter.
Follow the prompts to enter the folder path where you want to delete the desktop.ini files.

**Comment by sebhuet** [comment]
multithread script

import os
import threading
from queue import Queue
from concurrent.futures import ThreadPoolExecutor
import time

class DesktopIniCleaner:
    def __init__(self, max_workers=4):
        """
        Initialize the cleaner with a specified number of worker threads.
        
        Args:
            max_workers (int): Maximum number of concurrent threads
        """
        self.max_workers = max_workers
        self.files_deleted = 0
        self.errors = 0
        self.lock = threading.Lock()
        
    def delete_file(self, file_path):
        """
        Delete a single desktop.ini file and update counters.
        
        Args:
            file_path (str): Path to the file to delete
        """
        try:
            os.remove(file_path)
            with self.lock:
                self.files_deleted += 1
                print(f"Deleted {file_path}")
        except Exception as e:
            with self.lock:
                self.errors += 1
                print(f"Error deleting {file_path}: {e}")

    def collect_files(self, folder_path):
        """
        Collect all desktop.ini files in the given folder and its subfolders.
        
        Args:
            folder_path (str): Root folder to search
            
        Returns:
            list: List of paths to desktop.ini files
        """
        desktop_ini_files = []
        for root, _, files in os.walk(folder_path):
            for filename in files:
                if filename.lower() == "desktop.ini":
                    file_path = os.path.join(root, filename)
                    desktop_ini_files.append(file_path)
        return desktop_ini_files

    def clean_folder(self, folder_path):
        """
        Clean desktop.ini files from the specified folder using multiple threads.
        
        Args:
            folder_path (str): The folder to clean
            
        Returns:
            tuple: Number of files deleted and errors encountered
        """
        start_time = time.time()
        print(f"Scanning {folder_path} for desktop.ini files...")
        
        # First collect all the files
        desktop_ini_files = self.collect_files(folder_path)
        total_files = len(desktop_ini_files)
        
        print(f"Found {total_files} desktop.ini files. Starting deletion...")
        
        # Process files using thread pool
        with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
            executor.map(self.delete_file, desktop_ini_files)
        
        elapsed_time = time.time() - start_time
        
        return self.files_deleted, self.errors, elapsed_time

def main():
    folder_path = input("Enter the folder path: ")
    max_workers = int(input("Enter the number of worker threads (default 4): ") or "4")
    
    if not os.path.exists(folder_path):
        print("Error: The specified folder does not exist.")
        return
    
    cleaner = DesktopIniCleaner(max_workers)
    files_deleted, errors, elapsed_time = cleaner.clean_folder(folder_path)
    
    print("\nSummary:")
    print(f"Files deleted: {files_deleted}")
    print(f"Errors encountered: {errors}")
    print(f"Time taken: {elapsed_time:.2f} seconds")

if __name__ == "__main__":
    main()

**Comment by leov85** [comment]
Hi everyone,

I would like to share my simple solution I created from a suggestion found online.
It could be improved, but it has been working perfectly for my needs. 

I wrote a Windows batch file to automate a command that removes `desktop.ini` files from a folder and all its subdirectories. 

**Here is the code:**

```batch
@echo off
echo Current directory: %cd%
set /p "choice=Do you want to proceed with deleting desktop.ini files? (Y/N): "
if /i "%choice%"=="Y" (
    del desktop.ini /A:H /S
)
if /i "%choice%"=="N" (
    echo Execution aborted.
)
pause
```

**How to use it:**

1. Copy the code into a text file and name it whatever you like.
2. Change the file extension from `.txt` to `.bat`.
3. Place the `.bat` file in the root folder of your project.
4. Double-click the file to run it.
